hbase-server/src/main/java/org/apache/hadoop/hbase/procedure2/store/region/RegionProcedureStore.java

```diff
@@ -364,6 +364,7 @@ public void handleCorrupted(ProcedureIterator procIter) throws IOException {
       throw new IOException("Failed to delete the WALProcedureStore migrated proc wal directory " +
         procWALDir);
     }
+    store.stop(true);
```

Comment:
Do not need to be in finally block. If there are errors we just fail the start up of HMaster. If we pass abort = false then it will try to append a trailer which will cause teh stop to hang there.

Of course, a better way maybe call store.stop(false) before we delete the proc wal directory, but not a big deal?

Reply:
Ok. Good enough for now. Thanks.